src/main/java/dev/academy/patterns/abstractfactory/model/DataObjectFactory.java

```diff
@@ -0,0 +1,23 @@
+package dev.academy.patterns.abstractfactory.model;
+
+import dev.academy.patterns.abstractfactory.factory.AbstractFactory;
+import dev.academy.patterns.abstractfactory.factory.Factory;
+import dev.academy.patterns.abstractfactory.factory.GroupDataFactory;
+import dev.academy.patterns.abstractfactory.factory.UserDataFactory;
+
+import java.util.List;
+
+public class DataObjectFactory extends AbstractFactory<DataObject, String> {
+    @Override
+    public DataObject get(String json) {
+        return factories.stream()
+                .filter(factory -> factory.verify(json))
+                .findFirst()
+                .map(factory -> factory.fromJson(json))
```

Comment:
Please remember my monologue about pdf generation in my construction company and stuff. I guess the abstract factory should just return the factory. Otherwise you could extract `fromJson` as a convenience method.